incubator/hnc/test/conversion/conversion_test.go

```diff
@@ -0,0 +1,201 @@
+package conversion
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+
+	. "sigs.k8s.io/multi-tenancy/incubator/hnc/pkg/testutils"
+)
+
+var (
+	crds = []string{"subnamespaceanchors.hnc.x-k8s.io","hierarchyconfigurations.hnc.x-k8s.io", "hncconfigurations.hnc.x-k8s.io"}
+)
+
+func TestConversion(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	SetDefaultEventuallyTimeout(time.Second * 2)
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"HNC v1alpha2 Conversion Test Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = Describe("Conversion from v1alpha1 to v1alpha2", func() {
+	const (
+		// hncFromVersion has to be a version using v1alpha1.
+		hncFromVersion = "v0.5.0"
+		namspacePrefix = "e2e-conversion-test-"
+		nsA            = namspacePrefix + "a"
+		nsB            = namspacePrefix + "b"
+	)
+
+	BeforeEach(func() {
+		CleanupNamespaces(nsA, nsB)
+		// Tear down HNC for both the specified version and at HEAD.
+		TearDownHNC(hncFromVersion)
+		setupV1alpha1(hncFromVersion)
+	})
+
+	AfterEach(func() {
+		CleanupNamespaces(nsA, nsB)
+		// Only tear down HNC at HEAD, since that's what we just deployed.
+		TearDownHNC("")
+	})
+
+	It("should convert subnamespace anchors", func() {
+		// Before conversion, create namespace A and a subnamespace B.
+		MustRun("kubectl create ns", nsA)
+		anchor := `# temp file created by conversion_test.go
+apiVersion: hnc.x-k8s.io/v1alpha1
+kind: SubnamespaceAnchor
+metadata:
+  name: e2e-conversion-test-b
+  namespace: e2e-conversion-test-a`
+		MustApplyYAML(anchor)
+		RunShouldContain(nsB, 10, "kubectl get ns")
+		RunShouldContain("ok", 10, "kubectl get subnamespaceanchors.hnc.x-k8s.io", nsB, "-n", nsA, "-oyaml")
```

Comment:
Replace all numeric constants with named constants and documentation for why the values were chosen